README.md

```diff
@@ -18,8 +18,250 @@ Or install it yourself as:
+exceptions in a `YetiException::Error` and re-raise the new exception.  It adds
+the same attributes as `raise_exception`, but also does the following:
+
+- It maintains the original exception's backtrace.  This useful for identifying
+  where the exception was originally raised, not where it was wrapped and
+  reraised.
+
+- It includes the original exception's message in the `:msg` key of the details
+  hash.
+
+Note that this example uses the class method instead of the instance method to
+demonstrate both cases.
+
+    class OtherClass
+      ORIGINAL_ERROR = StandardError.new('Original error')
+
+      def self.call
+        raise ORIGINAL_ERROR
+      end
+    end
+
+    class MyClass
+      include YetiException::Helpers
+
+      def self.call
+        OtherClass.call
+      rescue => ex
+        wrap_exception(ex, YetiException::Error, { some: 'details' })
+      end
+    end
+
+    ex = begin
+      MyClass.call
+    rescue => exception
+      exception
+    end
+
+    ex.klass #=> MyClass
+    ex.details #=> {:some=>"details", :msg=>"Original error"}
+    ex.backtrace == OtherClass::ORIGINAL_ERROR.backtrace #=> true
+
+Again, a custom subclass and additional parameter can be used:
+
+    wrap_exception(ex,                        # original exception
+                   MyCustomError,             # exception class
+                   { some: 'details' },       # details
+                   false,                     # transient
+                   400)                       # HTTP status
+
+### Integration with [YetiLogger](https://github.com/Yesware/yeti_logger)
+
+`YetiException` was designed with `YetiLogger` in mind.  `YetiLogger`
+automatically handles logging a details hash with an optional exception.
+
+    begin
+      do_something
+    rescue YetiException::Error => ex
+      log_error(ex.details, ex)
+    end
+
+The class in which the exception was raised can also be included.
+
+    begin
+      do_something
+    rescue YetiException::Error => ex
+      log_error(ex.details.merge(klass: ex.klass), ex)
+    end
+
+### Integration with Ruby on Rails controllers
+
+The `http_status` attribute of `YetiException::Error` can be used when rescuing
+an exception in a Rails controller.  If your application uses `YetiException`
+for all errors, it makes sense to define a `rescue_from` hander in
+`ApplicationController`.  For example, for an API-only app:
+
+    class ApplicationController < ActionController::API
+
+      # Define a catch-all handler for other exceptions.  Rails searches for a
+      # matching handler in the reverse order of their definitions.
+      rescue_from StandardError do |ex|
+        render json: { error: ex.message }, status: 500
+      end
+
+      # For the YetiException exceptions with richer data, use the specified
+      # details and status.
+      rescue_from YetiException::Error do |ex|
+        render json: { error: ex.details }, status: ex.status
+      end
```

Comment:
[suggestion] Let's log the exceptions in these examples. I realize you showed logging examples above, but devs are going to use these snippets directly, and few things drive me crazier than unlogged errors.